server/channels/store/sqlstore/team_store.go

```diff
@@ -230,7 +230,7 @@ func (s SqlTeamStore) Save(team *model.Team) (*model.Team, error) {
 		return nil, err
 	}
 
-	if _, err := s.GetMasterX().NamedExec(`INSERT INTO Teams
+	if _, err := s.GetMasterX().NamedExecContext(ctx, `INSERT INTO Teams
```

Comment:
There are some problems with this. Directly setting the context from here means we are no longer setting the `context.WithTimeout` in `NamedExec` any more. That means this means gets to run without a timeout. Currently, we don't actually pass a context to the SQL query. All a context is used, is mainly to disinguish between executing a query in master vs replica.

Atleast, in this PR, all I am seeing is that we are just passing the context from `request.CTX` with no modifications. That is effectively useless at the moment because in the background, we are just using [context.Background](https://github.com/mattermost/mattermost/blob/2ed0c6495b011ae9d984b7eb45b97dd8f7bd2220/server/channels/web/handlers.go#L178). 

What was the real motivation behind this PR?

Reply:
I agree that we should preserve the original behavior of `NamedExec` by adding the timeout to the existing context. I see value on passing the context from outside, because is what I would expect from context in upper layers, for example, if I pass a timeout context, I don't mind if you also pass the store default timeout, but at least I want you to get my timeout too.

Reply:
> I agree that we should preserve the original behavior of `NamedExec` by adding the timeout to the existing context.

I agree with this and I didn't caught this when making changes. The idea was for `NamedExecContext` to create the context `WithTimeout` while `NamedExec` would call the previous method with `context.Background()` but I ended up doing the opposite.

I will wait for discussions to be resolved before making any more changes.

Reply:
Yeah I was mainly trying to understand the reasoning behind this exact PR. If this is just to kick off a community campaign to add context to all store methods, then I think we should do the following:
1. Move the app/request package to public/shared/request.
2. Change all context.Context in store layer to request.CTX.
3. Have an example PR like this which adds a request.CTX to a store method.

And then create a bunch of tickets (one per store method) in bunches.

Reply:
So the plan would be to address 1&2 on a single (or two) PRs, and then do 3 (which would replace this PR), right? Is there a JIRA for those first steps or is someone already handling them?

Reply:
Yes. No. :)